README.md

```diff
@@ -156,25 +161,50 @@ const urlFile = {
   method: 'GET'
 }
 
-const service = serviceBuilder
-  .setServiceEndpoint('https://ocean.provider.to/use') // the access controller to be in control of this asset
+serviceBuilder
+  .setServiceEndpoint('https://v4.provider.oceanprotocol.com') // v4 ocean provider, maintained by the Ocean Protocol Foundation
   .setTimeout(0) // Timeout in seconds (0 means unlimited access after purchase)
   .addFile(urlFile)
-  .build()
-
-assetBuilder.addService(service)
 ```
 
 > **_NOTE:_** If you want to publish an algorithm or dataset for computation make sure to set `ServiceBuilder(ServiceTypes.COMPUTE, ...)`.
 
-The code above will build a new `access` service, serving a `url` type file that is available at `https://link.to/my/asset`. The service will be accessible via the ocean provider hosted at `https://ocean.provider.to/use`.
+The code above will build a new `access` service, serving a `url` type file that is available at `https://link.to/my/asset`. The service will be accessible via the ocean provider hosted at `https://v4.provider.oceanprotocol.com`.
+
+> **_NOTE:_** You should specify the provider instance to use, as this is the access controlling component in the stack. You can be fully self-sovereign by using a self-deployed instance or use a trusted third party instance of your choice.
 
 The supported `ServiceTypes` are `ACCESS` and `COMPUTE`.
 <br />The supported `FileTypes` are `URL`, `GRAPHQL`, `ARWEAVE`, `IPFS` and `SMARTCONTRACT`.
 
 For more info on the different types and their available configuration, please refer to the official [Ocean Protocl Documentation](https://docs.oceanprotocol.com/core-concepts/did-ddo#files).
 
-### Consumer Parameters
+#### Pricing
+
+We also want to also specify the pricing for our asset. The `ServiceBuilder` provides a function for this that we can make use of:
+
+```ts
+// Example of a fixed asset
+serviceBuilder.setPricing({
+  type: 'fixed', // 'fixed' or 'free'
+  // freCreationParams can be ommitted for 'free' pricing schemas
+  freCreationParams: {
+    fixedRateAddress: '0x25e1926E3d57eC0651e89C654AB0FA182C6D5CF7', // Fixed Rate Contract address on Mumbai network
+    baseTokenAddress: '0xd8992Ed72C445c35Cb4A2be468568Ed1079357c8', // OCEAN token contract address on Mumbai network
+    baseTokenDecimals: 18, // adjusted to OCEAN token
+    datatokenDecimals: 18, // adjusted to OCEAN token
```

Comment:
todo: I think it's not referring to the OCEAN token.

Reply:
should be correct, see: https://github.com/oceanprotocol/contracts/blob/85a130307c7846cf354d8f919a52e0152d834906/addresses/address.json#L4C32-L4C32